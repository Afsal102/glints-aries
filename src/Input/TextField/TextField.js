import React, { Component } from 'react';

import Icon from '../../General/Icon';

import {
  TextFieldContainer,
  TextFieldInput,
  TextFieldLabel,
} from '../../Style/Input/TextFieldStyle';

class TextField extends Component <Props, State> {
  constructor(props) {
    super(props);
    const { type } = this.props;

    this.state = {
      floating: false,
      inputType: type,
    };
  }

  handleFocusChange = (onBlur) => {
    const listener = (e) => {
      this.setState({
        floating: e.target.value.length > 0,
      });

      if (onBlur !== undefined) {
        return onBlur();
      }
    };

    return listener;
  }

  handleShowPassword = () => {
    const { inputType } = this.state;

    this.setState({
      inputType: inputType === 'password' ? 'text' : 'password',
    });
  }

  componentDidMount() {
    const { value, defaultValue } = this.props;
    if (value !== undefined || defaultValue !== undefined) {
      if (value !== '') {
        this.setState({
          floating: true,
        });
      }
    }
  }

  static getDerivedStateFromProps(nextProps) {
    if (nextProps.value) {
      return {
        floating: true,
      };
    }
    return {
      floating: false,
    };
  }

  render() {
    const {
      type,
      label,
      value,
      status,
      disabled,
      className,
      onBlur,
      small,
      removeFloatingLabel,
      ...defaultProps
    } = this.props;

    const { floating, inputType } = this.state;

    return (
      <TextFieldContainer id="aries-textfield" className={className}>
        <TextFieldInput
          type={inputType}
          placeholder={removeFloatingLabel && label}
          status={status}
          disabled={disabled}
          onBlur={this.handleFocusChange(onBlur)}
          floating={floating}
          value={value}
          aria-label={label}
          small={small}
          {...defaultProps}
        />
        <If condition={!removeFloatingLabel}>
          <TextFieldLabel
            floating={floating}
            status={status}
            small={small}
          >
            {label}
          </TextFieldLabel>
        </If>
        {type === 'password'
          && (
            <div className="see-password" onClick={this.handleShowPassword}>
              <Icon
                name={inputType === 'password' ? 'eye' : 'eye-slashed'}
                color={inputType === 'password' ? 'black' : '#777777'}
                size="18"
              />
            </div>
          )
        }
      </TextFieldContainer>
    );
  }
}

type Props = {
  type: string,
  label: string,
  status: string,
  disabled: boolean,
  className: string,
  small: boolean,
  removeFloatingLabel: boolean,
}

type State = {
  floating: boolean,
};

export default TextField;
