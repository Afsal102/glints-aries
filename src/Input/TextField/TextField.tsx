import * as React from 'react';

import classNames from 'classnames';

import { EyeIcon, EyeSlashedIcon } from '../../General/Icon/components';

import {
  TextFieldContainer,
  TextFieldInput,
  TextFieldLabel,
} from '../../Style/Input/TextFieldStyle';

class TextField extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    const { type } = this.props;

    this.state = {
      floating: false,
      inputType: type,
    };
  }

  handleFocusChange = (
    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void
  ) => {
    const listener = (e: React.FocusEvent<HTMLInputElement>) => {
      this.setState({
        floating: e.target.value.length > 0,
      });

      if (onBlur !== undefined) {
        return onBlur(e);
      }
    };

    return listener;
  };

  handleShowPassword = () => {
    const { inputType } = this.state;

    this.setState({
      inputType: inputType === 'password' ? 'text' : 'password',
    });
  };

  handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const { disableTyping, onKeyDown } = this.props;

    if (disableTyping) {
      e.preventDefault();
    }

    if (onKeyDown !== undefined) {
      return onKeyDown(e);
    }
  };

  componentDidMount() {
    const { value, defaultValue } = this.props;
    if (value !== undefined || defaultValue !== undefined) {
      if (value !== '') {
        this.setState({
          floating: true,
        });
      }
    }
  }

  static getDerivedStateFromProps(nextProps: Props) {
    if (nextProps.value) {
      return {
        floating: true,
      };
    }
    return false;
  }

  render() {
    const {
      type,
      label,
      value,
      status,
      disabled,
      className,
      onBlur,
      small,
      removeFloatingLabel,
      disableTyping,
      ...defaultProps
    } = this.props;

    const { floating, inputType } = this.state;

    return (
      <TextFieldContainer className={classNames('aries-textfield', className)}>
        <TextFieldInput
          type={inputType}
          placeholder={removeFloatingLabel && label}
          status={status}
          disabled={disabled}
          onBlur={this.handleFocusChange(onBlur)}
          onKeyDown={this.handleKeyDown}
          floating={floating}
          value={value}
          aria-label={label}
          small={small}
          disableTyping={disableTyping}
          {...defaultProps}
        />
        {!removeFloatingLabel && (
          <TextFieldLabel
            data-testid="textfield-label"
            className="textfield-label"
            floating={floating}
            status={status}
            small={small}
          >
            {label}
          </TextFieldLabel>
        )}
        {type === 'password' && (
          <div className="see-password" onClick={this.handleShowPassword}>
            {inputType === 'password' ? (
              <EyeIcon color="black" />
            ) : (
              <EyeSlashedIcon color="#777777" />
            )}
          </div>
        )}
      </TextFieldContainer>
    );
  }
}

export type textFieldType = 'text' | 'password' | 'number';

interface Props extends React.ComponentPropsWithoutRef<typeof TextFieldInput> {
  type: textFieldType;
  label: string;
  disabled?: boolean;
  className?: string;
  removeFloatingLabel?: boolean;
}

interface State {
  floating: boolean;
  inputType: textFieldType;
}

export default TextField;
