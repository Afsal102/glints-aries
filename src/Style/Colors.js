export const PrimaryColor = {
  glintsred: '#EC272B',
  glintsblue: '#0BAEEC',
  glintsyellow: '#FFF240',
};

export const SecondaryColor = {
  white: '#FFFFFF',
  whitesmoke: '#F5F5F5',
  black: '#000000',
  lightblack: '#777777',
  grey: '#646464',
  lightgrey: '#C6C6C6',
  buttercup: '#F5A623',
  purplepizza: '#FF00E5',
  blue: '#0869C2',
  bluegem: '#9013FE',
};

