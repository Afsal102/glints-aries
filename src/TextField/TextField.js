import React, { Component } from 'react';
import Icon from '../General/Icon';
import {
  TextFieldContainer,
  TextFieldInput,
  TextFieldLabel,
} from '../Style/TextFieldStyle';

class TextField extends Component <Props, State> {
  constructor(props) {
    super(props);
    const { type } = this.props;

    this.state = {
      floating: false,
      inputType: type,
    };
  }

  handleFocusChange = (onBlur) => {
    const listener = (e) => {
      this.setState({
        floating: e.target.value.length > 0,
      });

      if (onBlur !== undefined) {
        return onBlur();
      }
    };

    return listener;
  }

  handleShowPassword = () => {
    const { inputType } = this.state;

    this.setState({
      inputType: inputType === 'password' ? 'text' : 'password',
    });
  }

  componentDidMount() {
    const { value } = this.props;
    if (value !== undefined) {
      if (value !== '') {
        this.setState({
          floating: true,
        });
      }
    }
  }

  render() {
    const {
      type,
      label,
      value,
      status,
      disabled,
      className,
      onBlur,
      ...defaultProps
    } = this.props;

    const { floating, inputType } = this.state;

    return (
      <TextFieldContainer className={className}>
        <TextFieldInput
          type={inputType}
          status={status}
          disabled={disabled}
          onBlur={this.handleFocusChange(onBlur)}
          floating={floating}
          value={value}
          {...defaultProps}
        />
        <TextFieldLabel floating={floating} status={status}>
          {label}
        </TextFieldLabel>
        {type === 'password'
          && (
            <div className="see-password" onClick={this.handleShowPassword}>
              <Icon
                name={inputType === 'password' ? 'eye' : 'eye-slashed'}
                color={inputType === 'password' ? 'black' : '#777777'}
                size="18"
              />
            </div>
          )
        }
      </TextFieldContainer>
    );
  }
}

type Props = {
  type: string,
  label: string,
  status: string,
  disabled: boolean,
  className: string,
}

type State = {
  floating: boolean,
};

export default TextField;
