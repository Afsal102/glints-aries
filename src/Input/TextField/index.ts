import TextField, { Props } from './TextField';

export { TextField, Props as TextFieldProps };

export default TextField;
