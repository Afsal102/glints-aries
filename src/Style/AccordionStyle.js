import styled, { keyframes } from 'styled-components';
import { PrimaryColor, SecondaryColor } from './Colors';

const animateItem = keyframes`
  from {
    opacity: 0;
    transform: translate3d(0, 100%, 0);
  }

  to {
    opacity: 1;
    transform: translate3d(0, 0, 0);
  }
`;

export const AccordionContainer = styled.div`
  border: 5px solid ${SecondaryColor.lightgrey};
  background-color: ${SecondaryColor.white};
`;

export const AccordionHeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 22px;
  font-weight: 600;
  color: ${({ isOpen }) => isOpen ? `${SecondaryColor.white}` : `${SecondaryColor.blue}`};
  background-color: ${({ isOpen }) => isOpen ? `${SecondaryColor.blue}` : null};
  padding: 1.8rem 1.5rem;

  &:hover {
    cursor: pointer;
    color: ${SecondaryColor.white};
    background-color: ${SecondaryColor.blue};
  }

  p {
    margin: 0;
    margin-right: 1em;
  }
`;

export const AccordionItemWrapper = styled.div`
  background: ${SecondaryColor.whitesmoke};
  position: relative;
  padding: 1.8rem 1.5rem;
  max-height: 250px;
  overflow-y: auto;
`;

export const AccordionItemList = styled.p`
  padding: .5rem 0;
  margin: 0;
  cursor: pointer;
  animation: ${animateItem} .3s linear;

  &:hover {
    color: ${PrimaryColor.glintsblue};
  }
`;

export const AccordionLabelWrapper = styled.div`
  padding: 1.8rem 1.5rem;
`;
