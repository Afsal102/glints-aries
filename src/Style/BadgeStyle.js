import styled from 'styled-components';
import { PrimaryColor, SecondaryColor } from './Colors';

export const Container = styled.div`
  display: inline-flex;
  background: ${PrimaryColor.glintsred};
  color: ${SecondaryColor.white};
  padding: .1rem .5rem;
  border-radius: 20px;
  font-size: 10px;
  font-weight: 600;
`;
