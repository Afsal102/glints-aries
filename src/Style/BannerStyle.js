import styled from 'styled-components';
import { PrimaryColor, SecondaryColor } from './Colors';
import { Theme } from '../Utils/StyleConfig';

/*
 * Wrapper
 */
export const BannerWrapper = styled.div`
  position: relative;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  width: 100%;
  height: 12em;
  background-image: url(${props => props.imgUrl}); 
`;

export const BannerLabel = styled.label`
  position: absolute;
  bottom: 0;
  left: 1.4em;
  text-transform: uppercase;
  font-weight: 600;
  padding: .2em 1em;
  color: ${SecondaryColor.white};
  letter-spacing: 1px;

  ${props => {
    switch (props.theme) {
      case `${Theme.BLUE}`:
        return`
          background-color: ${PrimaryColor.glintsblue};
        `;
      case `${Theme.RED}`:
        return`
          background-color: ${PrimaryColor.glintsred};
        `;
      case `${Theme.YELLOW}`:
        return`
          background-color: ${PrimaryColor.glintsyellow};
          color: ${SecondaryColor.black};
        `;
      default:
        return`
          background-color: ${SecondaryColor.black};
        `;
    }
  }};
`;