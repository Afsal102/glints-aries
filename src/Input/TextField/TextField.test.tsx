import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import TextField, { textFieldType } from './TextField';
import { SecondaryColor, PrimaryColor } from '../../Style/Colors';

const props = {
  type: 'text' as textFieldType,
  label: 'Username',
  onChange: jest.fn().mockImplementation(event => event.target.value),
};

const inputValue = 'glintsaries';

function setupTextField(otherProps?: any) {
  const { getByLabelText, ...utils } = render(
    <TextField
      type={props.type}
      label={props.label}
      onChange={props.onChange}
      {...otherProps}
    />
  );
  const textFieldInput = getByLabelText(props.label) as HTMLInputElement;
  return { textFieldInput, utils };
}

function setupTextFieldWithChangeEvent(otherProps?: any) {
  const { textFieldInput } = setupTextField(otherProps);
  userEvent.type(textFieldInput, inputValue);
  return { textFieldInput };
}

it('<TextField> should render with a label of Username and type of text', () => {
  const TextFieldSnapshot = renderer
    .create(<TextField type={props.type} label={props.label} />)
    .toJSON();
  expect(TextFieldSnapshot).toMatchSnapshot();
});

describe('when it is rendered', () => {
  it('should display the correct input label', () => {
    const { textFieldInput } = setupTextField();
    expect(textFieldInput).toBeTruthy();
  });

  it('should have the correct type', () => {
    const { textFieldInput } = setupTextField();
    expect(textFieldInput.type).toEqual(props.type);
  });

  it('should have an empty string as its value', () => {
    const { textFieldInput } = setupTextField();
    expect(textFieldInput.value).toEqual('');
  });
});

describe('when a value is entered', () => {
  it('should call onChange for each input', () => {
    setupTextFieldWithChangeEvent();
    expect(props.onChange).toHaveBeenCalledTimes(inputValue.length);
  });

  it('should return the input value when onChange is passed a function: event => event.target.value', () => {
    setupTextFieldWithChangeEvent();
    expect(props.onChange.mock.results.slice(-1)[0].value).toEqual(inputValue);
  });

  it('should display the correct value', () => {
    const { textFieldInput } = setupTextFieldWithChangeEvent();
    expect(textFieldInput.value).toEqual(inputValue);
  });
});

describe('when status is:', () => {
  it("'success', it should display a lightblack border", () => {
    const { textFieldInput } = setupTextField({ status: 'success' });
    expect(textFieldInput).toHaveStyle(`
      border-color: ${SecondaryColor.lightblack};
    `);
  });

  it("'error', it should display a red border", () => {
    const { textFieldInput } = setupTextField({ status: 'error' });
    expect(textFieldInput).toHaveStyle(`
      border-color: ${PrimaryColor.glintsred};
    `);
  });
});

describe('when disabled is true', () => {
  it('should display cursor: not-allowed', () => {
    const { textFieldInput } = setupTextField({ disabled: true });
    expect(textFieldInput).toHaveStyle('cursor: not-allowed;');
  });

  it('should be disabled', () => {
    const { textFieldInput } = setupTextField({ disabled: true });
    expect(textFieldInput).toBeDisabled();
  });
});

describe('when small is:', () => {
  it('true, it should display the correct size', () => {
    const { textFieldInput } = setupTextField({ small: true });
    expect(textFieldInput).toHaveStyle('padding: 13px 15px;');
  });

  it('false, it should display the correct size', () => {
    const { textFieldInput } = setupTextField();
    expect(textFieldInput).toHaveStyle('padding: 15px 20px;');
  });
});

describe('when disableTyping is true', () => {
  it('should display cursor: pointer', () => {
    const { textFieldInput } = setupTextField({ disableTyping: true });
    expect(textFieldInput).toHaveStyle('cursor: pointer;');
  });

  it('should not be able to input a value', () => {
    const { textFieldInput } = setupTextField({
      disableTyping: true,
    });
    userEvent.type(textFieldInput, inputValue);
    expect(textFieldInput.value).toBe('');
  });
});

describe('when removeFloatingLabel is true', () => {
  it('should remove the floating label', () => {
    const { textFieldInput } = setupTextField({ removeFloatingLabel: true });
    const textFieldLabel = textFieldInput.parentElement.querySelector(
      '[data-testid="textfield-label"]'
    ) as HTMLLabelElement;
    expect(textFieldInput.parentElement).not.toContainElement(textFieldLabel);
  });

  it('should display a placeholder', () => {
    const { utils } = setupTextField({ removeFloatingLabel: true });
    const textFieldInput = utils.getByPlaceholderText(props.label);
    expect(textFieldInput).toBeTruthy();
  });
});
