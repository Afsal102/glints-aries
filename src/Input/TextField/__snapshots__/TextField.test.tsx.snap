// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField> should render with a label of Username and type of text 1`] = `
<div
  className="aries-textfield sc-bdVaJa ktXGUd"
>
  <input
    aria-label="Username"
    className="sc-htpNat evWPXU"
    onBlur={[Function]}
    onKeyDown={[Function]}
    type="text"
  />
  <label
    className="textfield-label sc-bwzfXH gtXjtR"
    data-testid="textfield-label"
  >
    Username
  </label>
</div>
`;
